Pie Chart Controls 2

Drayz64 opened the pull request.

Drayz64 committed:
Changed the drop down in the line chart to update when modules are deleted/added. Also only modules with study sessions appear in the drop down

Added panels to move the drop downs in the line chart and pie chart to be below the back button

Updated the titles of all the charts to be more accurate

Made all the axis labels have the same bold font

Drayz64 committed:
Deleted FileFormatException

Drayz64 committed:
Merge branch 'main' of https://github.com/CM10313-Group-1/tempest into pie_chart

Drayz64 committed:
The warning message when deleting a module now only appears with modules that have study sessions

Simplified the DataProtectionPageTest by having it extend GUIHelper

Drayz64 committed:
Set an 80 character limit on the module name input field

Drayz64 committed:
Changed the InputPage interface to extend ActionListener - allowing the ActionButtonPanel to take in an InputPage ensuring the page using the action button panel has its own actionPerformed method.

This meant that Page no longer needed to implement ActionListener meaning that pages that need an ActionListener have to implement it - but in return pages that don't need an ActionListener don't have to implement it

Drayz64 committed:
Made getBackButton() in ActionButtonPanel return a BackButton instead of a JButton

Made state in Chart be protected instead of public

Made state in Supervisor private, and changed changed a variable name

Drayz64 committed:
Changed the Chart Controls page to only show modules that have study sessions

Drayz64 committed:
Forgot to add the if

Drayz64 committed:
Merge remote-tracking branch 'origin/pie_chart' into pie_chart

Drayz64 committed:
...

Drayz64 committed:
Added a backPanel with a back button inside to Page so that each separate page class doesn't need to create an instance of BackButton.

This also means that each page doesn't need a getBackButton method as this is found in Page

Drayz64 committed:
Added back button changes to ChartControlsPage

Drayz64 committed:
Converted ActionButtonPanel into EnterButton - because every class has a back button now the action button panel was unnecessary and so its now just an enter button

This allowed for tests to be slightly simplified and for no getActionButtonPanel() method in the 3 input pages

Drayz64 committed:
The Pie Chart and Line Chart now only show modules in the key that have study sessions

Drayz64 committed:
Updated the EnterButton JavaDoc

Drayz64 committed:
Fixed a line chart bug I made when changing the file earlier

Drayz64 committed:
Used manager.revalidate() to update the line chart and pie chart rather than needing to use resizeGUI() in GUIManager

Drayz64 committed:
Removed the unnecessary boolean uniqueName in AddModulePage because if the module name isn't unique you can just return from the method

Renamed some JavaDoc in GUIManager

Removed getCharts() in GUIManager as it's never used

Removed getPageName() in PageManager as it's never used

Drayz64 committed:
Turned InputPage interface into an abstract class to simplify input pages

Drayz64 committed:
Added a test in DeleteSessionPage to show the table changes when a session is deleted

Also move TextInputLimit into its own class file

Drayz64 committed:
Made the pageManager in GUIManager not be static

bridges-wood commented on src/main/java/tempest/ui/GUIManager.java:
```diff
@@ -178,8 +178,8 @@ public void resizeGUI() {
      *
-     * @param classObject A class extending page (e.g. HomePage.class)
-     * @return Page - The instance of the required class
+     * @param classObject A class extending Page (e.g. HomePage.class)
```
You could even use the Javadoc link here.

bridges-wood commented on src/main/java/tempest/ui/TextInputLimit.java:
```diff
@@ -0,0 +1,25 @@
+        this.limit = limit;
+    }
+
+    public void insertString (int offset, String  str, AttributeSet attr) throws BadLocationException {
```
Add some Javadoc here pls.

bridges-wood reviewed (changes requested):
Just some bits of Javadoc.

Drayz64 committed:
Removed TODO's in the chart pages and supervisor

Made two methods in CSVInterface private

Changed JavaDoc in GUIManager and TextInputLimit

Drayz64 committed:
Line Change
